network/address_test.go

```diff
@@ -63,3 +80,22 @@ func TestAddress(t *testing.T) {
 		assert.Equal(t, str.Public, add.Public())
 	}
 }
+
+// just a temporary test case
+func TestDNSNames(t *testing.T) {
+	assert.True(t, validHostname("myhost.secondlabel.org"), "valid")
+	assert.True(t, validHostname("www.asd.lol.xd"), "valid")
+	assert.True(t, validHostname("a.a"), "valid")
+	assert.True(t, validHostname("localhost"), "valid")
+	assert.True(t, validHostname("www.asd.lol.xd"), "valid")
+	assert.True(t, validHostname("randomtext"), "valid")
+
+	assert.False(t, validHostname("www.asd.lol.x-d"), "not valid")
+	assert.False(t, validHostname("192.168.1.1"), "not valid")
+	assert.False(t, validHostname("..a"), "not valid")
+	assert.False(t, validHostname("a..a"), "not valid")
+	assert.False(t, validHostname("123213.213"), "not valid") // look into this again
```

Comment:
Yes, for now I do not include IDNs (i.e. hyphens/numbers in TLDs are not admitted)